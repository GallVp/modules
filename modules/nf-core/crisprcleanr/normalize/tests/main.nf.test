
nextflow_process {

    name "Test Process CRISPRCLEANR_NORMALIZE"
    script "../main.nf"
    process "CRISPRCLEANR_NORMALIZE"

    tag "modules"
    tag "modules_nfcore"
    tag "crisprcleanr"
    tag "crisprcleanr/normalize"

    test("test-crisprcleanr-normalize") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/tsv/functional_genomics_counts.tsv', checkIfExists: true),
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/tsv/library_functional_genomics.tsv', checkIfExists: true)
				]
				input[1] = 30
				input[2] = 5

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.norm_count_file[0][1]).name,
                    process.out.versions
                    ).match()
                }
            )
        }
    }

    test("test-crisprcleanr-normalize-stub") {
        options '-stub'

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/tsv/functional_genomics_counts.tsv', checkIfExists: true),
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/tsv/library_functional_genomics.tsv', checkIfExists: true)
				]
				input[1] = 30
				input[2] = 5

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
