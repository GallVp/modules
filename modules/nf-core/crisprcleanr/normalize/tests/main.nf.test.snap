{
    "test-crisprcleanr-normalize": {
        "content": [
            "test_norm_table.tsv",
            [
                "versions.yml:md5,e954dc2bfa394dae2e55be2b1c16ae31"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-28T14:25:00.741578"
    },
    "test-crisprcleanr-normalize-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_norm_table.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e954dc2bfa394dae2e55be2b1c16ae31"
                ],
                "norm_count_file": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_norm_table.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e954dc2bfa394dae2e55be2b1c16ae31"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-28T14:34:59.649477"
    }
}